xformers/triton/k_fused_matmul_fw.py

```diff
@@ -0,0 +1,194 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All rights reserved.
+#
+# This source code is licensed under the BSD license found in the
+# LICENSE file in the root directory of this source tree.
+
+from typing import Optional
+
+import torch
+import triton
+import triton.language as tl
+
+# CREDITS: Initially inspired by the Triton tutorial on matrix multiplications
+
+
+# fmt: off
+@triton.heuristics({
+    'EVEN_K': lambda *args, **meta: args[7] % (meta['BLOCK_K']) == 0,
+})
+@triton.autotune(
```

Comment:
Just curious, are these values from testing and seeing which configs help?

Reply:
ah I need to explain that, it's really hard to grasp.. args[7] relates to the kernel inputs below, it's just the dimension "K" (the dimension against which the matrix mul is consolidated). This "heuristic" decorator states that if K is divisible by BLOCK_K (triton which test out the presets above for this value, best will depend on HW and inputs), then there's no need to mask the loads in the kernels (this `tl.load(.., maskl=())`), we will not overreach anyway.